test/testenv/appframework_utils.go

```diff
@@ -12,3 +16,64 @@ func GenerateAppSourceSpec(appSourceName string, appSourceLocation string, appSo
 		AppSourceDefaultSpec: appSourceDefaultSpec,
 	}
 }
+
+// GetPodAppStatus Get the app install status and version number
+func GetPodAppStatus(deployment *Deployment, podName string, ns string, appname string) (string, string, error) {
+	_, output := GetPodAppInstallStatus(deployment, podName, ns, appname)
+	status := strings.Fields(output)[1]
+	version, err := GetPodInstalledAppVersion(deployment, podName, ns, appname)
+	return status, version, err
+
+}
+
+// GetPodInstalledAppVersion Get the version of the app installed on pod
+func GetPodInstalledAppVersion(deployment *Deployment, podName string, ns string, appname string) (string, error) {
+	path := "etc/apps"
+	if strings.Contains(podName, "-indexer-") {
+		path = "etc/slave-apps/_cluster/default"
+	}
+	filePath := fmt.Sprintf("/opt/splunk/%s/%s/default/app.conf", path, appname)
+
+	confline, err := GetConfLineFromPod(podName, filePath, ns, "Version", "", false)
+	if err != nil {
+		logf.Log.Error(err, "Failed to get Version from pod", "Pod Name", podName)
+		return "", err
+	}
+	version := strings.TrimSpace(strings.Split(confline, " ")[2])
+	return version, err
+
+}
+
+// GetPodAppInstallStatus Get the app install status
```

Comment:
Would be a good idea to return err as well. That way caller decides how to fail. Return `false` when command fails is incorrect.

Reply:
modified to return err instead of boolean